### ISSUE TITLE
In `Obligation num : type`, `type` is ignored.

### ISSUE THREAD

**Issue by Zimmi48** [coq/coq#16839]
#### Description of the problem

Example:

```coq
Require Import Program Arith.

Program Definition id (n : nat) : { x : nat | x = n } :=
  if dec (leb n 0) then 0
  else S (pred n).
Obligation 1 : True.
```

Found by looking at the code:
https://github.com/coq/coq/blob/b9a1f1e5c7e41c6a32c968b5d085d3f2273e2f29/vernac/declare.ml#L2390-L2399

Don't we raise warnings for unused variables?

We should either check the type or remove the "feature" from the grammar. It was not documented before we fixed the grammar for this chapter (but this happened in 8.12).

#### Coq Version

8.16-8.17

**Comment by SkySkimmer** [comment]
>Don't we raise warnings for unused variables?

If you mean ocaml warnings then this variable is considered "innocuous" and needs warning 27 instead of 26 which is the one we use.

**Comment by Zimmi48** [comment]
Yes, I meant OCaml warning. Is there a reason why we did not activate warning 27? Is there some reference list of all the OCaml warnings?

**Comment by SkySkimmer** [comment]
Warning 27 was a bit painful when I tried it. We do use it in the kernel though.
We used to be more explicit about our warnings but currently we use dune defaults with a few changes, see /dune file (except in kernel/). Not sure why we changed.

**Comment by SkySkimmer** [comment]
I do `man ocamlc` and search a double digit number to find the doc warning list
